Open-Rival/include/TimerUtils.h

```diff
@@ -1,11 +1,30 @@
 #ifndef TIMER_UTILS_H
 #define TIMER_UTILS_H
 
+#include <chrono>
+
 namespace Rival {
     namespace TimerUtils {
 
         static const int timeStepMs = 1000 / 60;
 
+        using MillisecondsType = std::chrono::milliseconds;
+        using MillisecondsRep = decltype(MillisecondsType());
+        using TimeType = decltype(std::chrono::system_clock::now());
+
+        static const auto gameStartTime(
+            std::chrono::duration_cast<MillisecondsType>(
+                std::chrono::system_clock::now().time_since_epoch()));
+        MillisecondsType getCurrentTime();
+
+        // Return the time passed from t1 to t2
+        MillisecondsType getTimeDifference(TimeType t1, TimeType t2);
+
+        // Return the time passed since t
+        MillisecondsType getTimeSince(MillisecondsType t);
+
+        MillisecondsRep toMS(uint32_t t);
```

Comment:
Why do we need `MillisecondsRep`? Why can't this just return `MillisecondsType`?

Reply:
It could be. Then we would store a std:ratio along with MillisecondsRep in each UnitAnimationLookup, which would not be a "big deal".

Reply:
I'm not sure what you mean! I don't think I understand the difference between `MillisecondsRep` and `MillisecondsType`.

Reply:
Oops. Checking that, I noticed my two mistakes. I intended

> using MillisecondsRep = decltype(MillisecondsType(0).count());

which would be the underlying type (long in the example below) when we do MillisecondsType(0).count(). That was first mistake. Second was I thought storing just the MillisecondsRep would occupy less space, since it is not MillisecondsType, which (I thought) stores also a std::ratio for conversions.
See [this link](https://godbolt.org/z/WzL62y)
I was wrong about both. Hence what you are saying is correct. Could not check with MSVC but clang gave the same results.

Reply:
That makes more sense now :D